include/oneapi/dpl/pstl/hetero/dpcpp/unseq_backend_sycl.h

```diff
@@ -185,15 +185,67 @@ struct __init_processing
 //------------------------------------------------------------------------
 
 template <typename _ExecutionPolicy, typename _Operation1, typename _Operation2>
-struct transform_init
+struct transform_init_seq
 {
     _Operation1 __binary_op;
     _Operation2 __unary_op;
 
-    template <typename _NDItemId, typename _Size, typename _AccLocal, typename... _Acc>
+    template <typename _Size, typename _AccLocal, typename... _Acc>
     void
-    operator()(const _NDItemId __item, _Size __n, ::std::size_t __iters_per_work_item, ::std::size_t __global_id,
-               ::std::size_t __global_offset, _AccLocal& __local_mem, const _Acc&... __acc) const
+    operator()(const _Size __n, _AccLocal& __local_mem, const _Acc&... __acc) const
+    {
+        __local_mem = __unary_op(0, __acc...);
+        // Add neighbour to the current __local_mem
+        for (::std::size_t __i = 1; __i < __n; ++__i)
+            __local_mem = __binary_op(__local_mem, __unary_op(__i, __acc...));
+    }
+};
+
+template <typename _ExecutionPolicy, ::std::size_t __iters_per_work_item, typename _Operation1, typename _Operation2>
+struct transform_init_known
+{
+    _Operation1 __binary_op;
+    _Operation2 __unary_op;
+
+    // __iters_per_work_item2 is only needed to unify the interface for when __iters_per_work_item is unknown
+    template <typename _Size, typename _AccLocal, typename... _Acc>
+    void
+    operator()(const ::std::size_t __local_id, const _Size __n, const ::std::size_t __iters_per_work_item2,
+               const ::std::size_t __global_id, const ::std::size_t __global_offset, _AccLocal& __local_mem,
+               const _Acc&... __acc) const
+    {
+        const ::std::size_t __adjusted_global_id = __global_offset + __iters_per_work_item * __global_id;
+        const ::std::size_t __items_to_process = __n - (__iters_per_work_item * __global_id);
+        if (__items_to_process >= __iters_per_work_item)
+        {
+            typename _AccLocal::value_type __res = __unary_op(__adjusted_global_id, __acc...);
```

Comment:
This seems to result in a worse device code breaking the wide memory alignment.

Reply:
This is interesting.  It makes sense to add a comment here explaining that so that others are aware and to ensure it remains through future touches of the code.